src/diagnostics/MOM_spatial_means.F90

```diff
@@ -300,23 +296,26 @@ function global_volume_mean(var, h, G, GV, scale, tmp_scale, unscale)
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
 
   temp_scale = 1.0 ; if (present(tmp_scale)) temp_scale = tmp_scale
-  scalefac = temp_scale
+
+  scalefac = 1.0
```

Comment:
The assignment of `scalefac` seems problematic here. Whether `scalefac` includes `temp_scale` entirely depends on if `scale`/`unscale` is present. This decides whether `temp_scale` should be included in the `reproducing_sum` call in line 310. 

If we follow the convention in subroutine `global_layer_mean`, in which case `scalefac` always includes `temp_scale`, then `temp_scale` should not be included in line 310. Alternatively, `temp_scale` can be excluded from `scalefac` and explicitly used in `reproducing_sum` calls (probably better).

Reply:
Very good point.  I am modifying the code to avoid conflating `scalefac` and `temp_scale`.